s2a_options.go

```diff
@@ -120,6 +120,8 @@ type ClientOptions struct {
 	//
 	// If true, enables the use of S2Av2.
 	EnableV2 bool
+	// If true, enables the use of legacy S2Av1.
+	EnableLegacyMode bool
```

Comment:
Should we add logic to explicitly disable v2 if EnableLegacyMode is set to true?

Reply:
SGTM.